modin/experimental/engines/omnisci_on_native/frame/omnisci_worker.py

```diff
@@ -175,13 +170,43 @@ def put_arrow_to_omnisci(cls, table, name=None):
                     field.name, pa.string(), field.nullable, field.metadata
                 )
                 new_schema = new_schema.set(i, new_field)
+            # OmniSci doesn't support importing Arrow's date type:
+            # https://github.com/omnisci/omniscidb/issues/678
```

Comment:
Added the following comment:
[Arrow's date](https://arrow.apache.org/docs/python/generated/pyarrow.date32.html) is the number of days since the UNIX-epoch, so we can convert it to a timestamp[s] (number of seconds since the UNIX-epoch) without losing precision.